### DISCUSSION TITLE
Neither the MATCH_PASSWORD environment variable nor the local keychain contained a password.

### DISCUSSION THREAD

**Discussion by medaminenajjar** [fastlane/fastlane#16915] [Q&A]
In the execute shell in Jenkins i put this:
`fastlane run match username:"*******" git_url:"*******" clone_branch_directly:false team_id:"*******" team_name:"*******" type:"appstore" app_identifier:"*******" `

In the fastfile i put this:
```
lane :beta do
  build_app(scheme: "RandoGard")
  upload_to_testflight
end
```

And in the appfile i put this:
```
git_url "***********"
type "appstore" 
username "***********"
team_id "***********" 
team_name "***********"
apple_id "***********"
app_identifier '***********'
shallow_clone true
force_for_new_devices true
ENV["FASTLANE_PASSWORD"] = "***********"
ENV["MATCH_PASSWORD"] = "***********"
```

When i run the build via Jenkins it display :
```
[14:21:22]: [31mNeither the MATCH_PASSWORD environment variable nor the local keychain contained a password.[0m
[14:21:22]: [31mBailing out instead of asking for a password, since this is non-interactive mode.[0m
[14:21:22]: [31mCouldn't decrypt the repo, please make sure you enter the right password![0m
security: SecKeychainSearchCopyNext: The specified item could not be found in the keychain.
```

**Comment by AustinFuller** [comment]
I'm having the same error for different reasons. 

Related to your case, my best guess is that the `ENV["MATCH_PASSWORD"] = "***********"` isn't actually setting the environment variable. Try setting that environment variable more directly in Jenkins rather than within the appfile.

**Reply by rogerluan** [reply]
☝️What he said☝️ 👍 

When doing that, the env var name should be `MATCH_PASSWORD`
